### PR TITLE
SDC catalog app - Added behaviors tab

### LINKED ISSUE #1524: SDC catalog - Add behavior tab (to show skip logic and calculations behavior)
**Is your feature request related to a problem? Please describe.**
We need an additional tab to demonstrate SDC behavior such as calculations and skip logic

**Describe the solution you'd like**
Add a 3rd tab to the navigation bar. 

**Describe alternatives you've considered**
@jingtang10 and I have discussed adding this to "components", but behavior feels distinct of how multiple components might behave and interact. It also doesn't fit in layouts. 

**Additional context**
See figma file posted in chat. 
SDC - landing - behaviors

### PR DESCRIPTION
**Pull request by PallaviGanorkar**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1524

**Description**
Added Behaviors tab in Catalog apps bottom navigation bar having Calculation and Skip logic UI

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: (Bug fix | Feature | Documentation | Testing | Code health | Builds | Releases | Other)

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Review by santosh-pingle — COMMENTED** [review]
Can you please provide the short summary of implementation and video of implemented feature?
once done please reassign back to me.

**Comment by PallaviGanorkar** [comment]
@santosh-pingle icons in bottom navigation menu was not finalized, so could not add screen shot. Now Icons are updated please check in screen shot. Also Please do first round of review. Thanks

Behavior_tab_added

File: catalog/src/main/java/com/google/android/fhir/catalog/BehaviorListFragment.kt [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    super.onViewCreated(view, savedInstanceState)
+    setUpBehaviorsRecyclerView()
+  }
+
+  override fun onResume() {
+    super.onResume()
+    setUpActionBar()
+    (requireActivity() as MainActivity).showBottomNavigationView(View.VISIBLE)
+  }
+  private fun setUpActionBar() {
```
Comment (jingtang10) [inline catalog/src/main/java/com/google/android/fhir/catalog/BehaviorListFragment.kt:40]: add an empty line between the functions

File: catalog/src/main/java/com/google/android/fhir/catalog/BehaviorListFragment.kt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    (activity as MainActivity).setActionBar(
+      getString(R.string.toolbar_text),
+      Gravity.CENTER_HORIZONTAL
+    )
+    setHasOptionsMenu(true)
+  }
+  private fun setUpBehaviorsRecyclerView() {
```
Comment (jingtang10) [inline catalog/src/main/java/com/google/android/fhir/catalog/BehaviorListFragment.kt:47]: same here

File: catalog/src/main/res/navigation/nav_graph.xml [hunk 3]
```diff
@@ -47,10 +51,35 @@
...
+            app:destination="@id/componentListFragment"
+        />
+    <action
+            android:id="@+id/action_behaviorsFragment_to_layoutsFragment"
+            app:destination="@id/layoutListFragment"
+        />
+    <action
+            android:id="@+id/action_behaviorsFragment_to_galleryQuestionnaireFragment"
+            app:destination="@id/galleryQuestionnaireFragment"
+        />
```
Comment (jingtang10) [inline catalog/src/main/res/navigation/nav_graph.xml:80]: so if we have `n` tabs, adding another tab will always require another `2*n` additional actions? this seems really inefficient - is there a better way of doing this or this is just now navigation graph works?

**Review by jingtang10 — APPROVED** [review]

File: catalog/src/main/java/com/google/android/fhir/catalog/BehaviorListFragment.kt (hunk 2 above)
Comment (PallaviGanorkar) [reply catalog/src/main/java/com/google/android/fhir/catalog/BehaviorListFragment.kt:47]: Done

File: catalog/src/main/res/navigation/nav_graph.xml (hunk 3 above)
Comment (PallaviGanorkar) [reply catalog/src/main/res/navigation/nav_graph.xml:80]: Action represent that path to Destination fragment in nav grap, as per my finding, which can not be optimized. Also Maximum number of items supported by BottomNavigationView is 5

**Pull request merged**